vm/vm.go

```diff
@@ -102,6 +102,46 @@ func matchesAssetParams(newAsset, prevAsset *asset.Asset,
 	return nil
 }
 
+// extractSighashFromWitnessStack attempts to extract the sighash flag from a
+// witness stack. If no sighash flag is found, the default sighash flag is
+// returned.
+func extractSighashFromWitnessStack(
```

Comment:
I think we should eliminate this function, as it only assumes a keyspend sighash. A valid witness may have > 1 signature (multi-sig, etc) in the witness (script path).